src/test.rs

```diff
@@ -362,34 +359,113 @@ fn it_works_with_lots_of_things_codec() {
 
 #[test]
 fn it_works_with_validation() {
-    let method = Method {
-        name: "method_test".to_string(),
-        block_cooldown: Some(123),
-        constraints: None,
-    };
-
-    let mut methods: Vec<(String, Method)> = Default::default();
-    methods.push((method.name.clone(), method.clone()));
+    {
+        let contract = Contract {
+            data_table: DataTable::new(vec![
+                PactType::Numeric(Numeric(111)),
+                PactType::Numeric(Numeric(333)),
+                PactType::StringLike(StringLike(b"testing")),
+            ]),
+            bytecode: [OpCode::EQ.into(), 0, 0, 1, 0, OpCode::EQ.into(), 0, 1, 1, 1].to_vec(),
+        };
+        let mut constraints: Vec<u8> = Vec::new();
+        contract.encode(&mut constraints);
+
+        let method = Method {
+            name: "method_test".to_string(),
+            block_cooldown: Some(123),
+            constraints: Some(constraints.clone()),
+        };
+
+        let mut methods: Vec<(String, Method)> = Default::default();
+        methods.push((method.name.clone(), method.clone()));
+
+        let module = Module {
+            name: "module_test".to_string(),
+            block_cooldown: Some(86_400),
+            methods: methods.clone(),
+        };
+        let mut modules: Vec<(String, Module)> = Default::default();
+        modules.push((module.name.clone(), module.clone()));
+
+        let cennznut = CENNZnutV0 { modules };
+        let args = [
+            PactType::Numeric(Numeric(0)),
+            PactType::StringLike(StringLike(b"test")),
+        ];
+        assert_eq!(cennznut.validate(&module.name, &method.name, &args), Ok(()));
```

Comment:
This test should be failing as `args` doesn't meet the conditions in `contract` so should return `Ok(false)`